src/game_battlealgorithm.cpp

```diff
@@ -1304,39 +1303,41 @@ bool Game_BattleAlgorithm::Skill::Execute() {
 		}
 
 		// Conditions:
-		bool heals_states = IsPositiveSkill() ^ (Player::IsRPG2k3() && skill.reverse_state_effect);
-		for (int i = 0; i < (int) skill.state_effects.size(); i++) {
-			if (!skill.state_effects[i])
-				continue;
-			auto state_id = i + 1;
+		// If the target gets killed by damage, do not add or remove states
+		if (!lethal) {
+			bool heals_states = IsPositiveSkill() ^ (Player::IsRPG2k3() && skill.reverse_state_effect);
+			for (int i = 0; i < (int) skill.state_effects.size(); i++) {
```

Comment:
Change to `static_cast<int>()`. Some compilers warn about C casts

Reply:
Did the change and updated this PR.